napari/layers/shapes/_shapes_models/_polygon_base.py

```diff
@@ -5,6 +5,15 @@
 from napari.layers.shapes._shapes_utils import create_box
 from napari.utils.translations import trans
 
+try:
+    from napari.layers.shapes._accelerated_triangulate import (
+        remove_path_duplicates,
+    )
+except ImportError:
+
+    def remove_path_duplicates(data: np.ndarray, closed: bool):
+        return data
```

Comment:
@Czaki do we not need to remove duplicates when there is no numba? I can get pretty similar performance with a NumPy version:

```python
def remdup(p):
    diff = np.diff(np.append(p, p[0:1, :], axis=0), axis=0)
    dup = np.all(diff == 0, axis=1)
    indices = np.arange(p.shape[0])
    return p[indices[~dup]]
```

This takes 65µs on a 100 element array with 10 duplicates, vs 50µs on the same array with numba.

Reply:
(yes I know that we need some more checks but probably not the bottleneck)

Reply:
This code does not work with repeated points on begin/end of a closed path.

Reply:
Here's the final func I just pushed, which handles both the lasso duplicate points and closed paths:

```python
def _remove_path_duplicates_np(data: np.ndarray, closed: bool):
    # We add the first data point at the end to get the same length bool
    # array as the data, and also to work on closed shapes; the last value
    # in the diff array compares the last and first vertex.
    diff = np.diff(np.append(data, data[0:1, :], axis=0), axis=0)
    dup = np.all(diff == 0, axis=1)
    # if the shape is closed, check whether the first vertex is the same
    # as the last vertex, and count it as a duplicate if so
    if closed and dup[-1]:
        dup[0] = True
    # we allow repeated nodes at the end for the lasso tool, which
    # for an instant needs both the last placed point and the point at the
    # cursor to be the same; if the lasso implementation becomes cleaner,
    # remove this hardcoding
    dup[-2:] = False
    indices = np.arange(data.shape[0])
    return data[indices[~dup]]
```